### PR TITLE
Add static robot information in local db

### PR DESCRIPTION
**Pull request by kvsikand**
Adds static robot information in a local `levelDb` instance.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,16 +16,23 @@
...
+    "@types/leveldown": "^4.0.2",
+    "@types/levelup": "^4.3.0",
     "cors": "^2.8.5",
     "express": "^4.17.1",
     "flatbuffers": "^1.12.0",
     "google-auth-library": "^6.0.2",
     "ip6addr": "^0.2.3",
+    "leveldown": "^5.6.0",
+    "levelup": "^4.4.0",
+    "mongoose": "^5.10.2",
```
Comment (kvsikand) [inline package.json:28]: oops, removing unnecessary dependencies

File: src/status.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+   * @param name string name of the client sending the message
+   */
+  handleMessageBuffer(sender: WebSocket, buf: flatbuffers.ByteBuffer, clientIp: string, topic: string, logger?: Logger) {
+    if (!this.connected) {
+      this.logger?.log(`Not connected to database`);
+      return;
+    }
+
+    const metadata = getRobofleetMetadata(buf);
+    if (metadata?.type() === "amrl_msgs/RobofleetStatus" || metadata?.type() === "RobofleetStatus" ) {
```
Comment (loganzartman) [inline src/status.ts:107]: technically the type `RobofleetStatus` (as opposed to `amrl_msgs/RobofleetStatus`) is not valid; we should open an issue to fix this if we use it anywhere.

**Review by loganzartman — COMMENTED** [review]
awesome, thanks for doing this! I think there's some dead code we can remove, but otherwise seems reasonable to me

File: src/expressApp.ts [hunk 3]
```diff
@@ -6,10 +6,22 @@ import config from "./config";
...
+  async function handleDbConnection() {
+    // populate known clientNames
+    console.log("Express App: connected to database");
+  }
+  
+  function handleDbConnError() {
+    // idk, log stuff?
+    console.log("Express App: error connecting to database");
+  }
+
```
[the remark this answers is not in the document]
Comment (kvsikand) [reply src/expressApp.ts:24]: Yup, thanks

File: src/schema_generated.ts [hunk 4]
```diff
@@ -3235,6 +3235,115 @@ static endPose(builder:flatbuffers.Builder):flatbuffers.Offset {
...
+ * @param flatbuffers.Builder builder
+ * @returns flatbuffers.Offset
+ */
+static endPoseStamped(builder:flatbuffers.Builder):flatbuffers.Offset {
+  var offset = builder.endObject();
+  builder.requiredField(offset, 6); // header
+  builder.requiredField(offset, 8); // pose
+  return offset;
+};
+
```
[the remark this answers is not in the document]
Comment (kvsikand) [reply src/schema_generated.ts:3346]: oh yeah, definitely unrelated.

File: src/status.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      // set up intermittent saves to the db for each robot
+      setInterval(this.saveClientInformation.bind(this), StatusManager.CLIENT_SAVE_INTERVAL);
+      this.connected = true;
+    });
+  }
+
+  handleDbConnError() {
+    // idk, log stuff?
+    this.logger?.log("Error connecting to database");
+  }
```
[the remark this answers is not in the document]
Comment (kvsikand) [reply src/status.ts:42]: Yup

File: src/status.ts (hunk 2 above)
Comment (kvsikand) [reply src/status.ts]: This was an unnecessary part of the the conditional, I'll drop the second part.

**Pull request merged**